### PR TITLE
Create decode_tx_input.rs

### PR DESCRIPTION
**Pull request by elyase**
## Motivation
The question comes up often in the Telegram chat

## Solution
Provided an example

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Updated the changelog

### CODE REVIEW

File: examples/decode_tx_input.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    IUniswapV2Router02,
+    r#"[
+        swapExactTokensForTokens(uint256 amountIn, uint256 amountOutMin, address[] calldata path, address to, uint256 deadline)
+    ]"#,
+);
+fn main() -> Result<()> {
+    println!("Decoding https://etherscan.io/tx/0xd1b449d8b1552156957309bffb988924569de34fbf21b51e7af31070cc80fe9a");
+    let tx_input = "0x38ed173900000000000000000000000000000000000000000001a717cc0a3e4f84c00000000000000000000000000000000000000000000000000000000000000283568400000000000000000000000000000000000000000000000000000000000000a0000000000000000000000000201f129111c60401630932d9f9811bd5b5fff34e000000000000000000000000000000000000000000000000000000006227723d000000000000000000000000000000000000000000000000000000000000000200000000000000000000000095ad61b0a150d79219dcf64e1e6cc01f0b64c4ce000000000000000000000000dac17f958d2ee523a2206206994597c13d831ec7";
+    let calldata = tx_input.strip_prefix("0x").unwrap_or(tx_input);
+    let calldata = Bytes::from(hex::decode(calldata).unwrap());
```
Comment (mattsse) [inline examples/decode_tx_input.rs:18]: I believe you can simply do
```suggestion
    let calldata: Bytes = input.parse().unwrap();
```
here

File: examples/decode_tx_input.rs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    let decoded = SwapExactTokensForTokensCall::decode(&calldata)?;
```
Comment (mattsse) [inline examples/decode_tx_input.rs:19]: this makes me think wether we should add a `decode_hex` function?

cc @gakonst

**Review by mattsse — CHANGES_REQUESTED** [review]
mind trying my suggestion, pretty sure this works

File: examples/decode_tx_input.rs (hunk 2 above)
Comment (gakonst) [reply examples/decode_tx_input.rs:19]: Yeah `decode_hex(data: &str)` which does the conversion internally, is blanked impl'ed on the trait and forwards to `decode` seems right

File: examples/decode_tx_input.rs (hunk 1 above)
Comment (elyase) [reply examples/decode_tx_input.rs]: nice! added the suggestion.

**Pull request merged**